### PR TITLE
Remove IApplicationEnvironment and IRuntimeEnvironment

### LINKED ISSUE #36: Consider removing IApplicationEnvironment \ IRuntimeEnvironment
* Runtime components in Asp.Net have switched to using `IHostingEnvironment` over `IApplicationEnvironment`. 
* CoreFX has a `RuntimeInformation` type that has most of the API that `IRuntimeEnvironment` supports.

Given these, we could remove these types from our codebase. We can follow up with a change to the CLI so that it removes one arc of the cyclic dependency that is currently present between the two.

cc @davidfowl  \ @Eilon

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #36

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @davidfowl \ @Eilon

File: src/Microsoft.Extensions.PlatformAbstractions/PlatformServices.cs [hunk 1]
```diff
@@ -3,25 +3,18 @@
...
+            Application = new ApplicationEnvironment();
+            Runtime = new RuntimeEnvironment();
         }
 
-        public abstract IApplicationEnvironment Application { get; }
+        public static PlatformServices Default { get; } = new PlatformServices();
 
-        public abstract IRuntimeEnvironment Runtime { get; }
-        
-        public static void SetDefault(PlatformServices defaultPlatformServices)
```
Comment (pranavkm) [inline src/Microsoft.Extensions.PlatformAbstractions/PlatformServices.cs]: Do we want to keep this around? It's kinda useless now that you can no longer create `IApplicationEnvironment`, \ `IRuntimeEnvironment` to make this useful.

File: src/Microsoft.Extensions.PlatformAbstractions/RuntimeEnvironmentExtensions.cs [hunk 2]
```diff
@@ -7,15 +7,15 @@ public static class RuntimeEnvironmentExtensions
         private static readonly string OverrideEnvironmentVariableName = "DOTNET_RUNTIME_ID";
         private static readonly string LegacyOverrideEnvironmentVariableName = "DNX_RUNTIME_ID";
```
Comment (pranavkm) [inline src/Microsoft.Extensions.PlatformAbstractions/RuntimeEnvironmentExtensions.cs]: Can we remove this? cc @anurse

Comment (moozzyk) [inline src/Microsoft.Extensions.PlatformAbstractions/RuntimeEnvironmentExtensions.cs]: I am not even sure if `DOTNET_RUNTIME_ID` works anymore end to end. I tried it on Debian some time ago and it did not help (modifying /etc/os_release did help)

**Pull request merged**